### PR TITLE
AP_MSP: fix for multiple backends initialization

### PR DESCRIPTION
**Pull request by yaapu**
A fix for multiple MSP backends initialization.

related issue https://github.com/ArduPilot/ardupilot/issues/15756

### CODE REVIEW

File: libraries/AP_MSP/AP_MSP.cpp [hunk 1]
```diff
@@ -102,7 +102,7 @@ void AP_MSP::init()
     for (uint8_t protocol_instance=0; protocol_instance<MSP_MAX_INSTANCES-_msp_status.backend_count; protocol_instance++) {
```
Comment (khancyr) [inline libraries/AP_MSP/AP_MSP.cpp:102]: insn't the issue just : 
`for (uint8_t protocol_instance=_msp_status.backend_count; protocol_instance<MSP_MAX_INSTANCES; protocol_instance++)`

File: libraries/AP_MSP/AP_MSP.cpp [hunk 2]
```diff
@@ -86,7 +86,7 @@ void AP_MSP::init()
     for (uint8_t protocol_instance=0; protocol_instance<MSP_MAX_INSTANCES; protocol_instance++) {
```
Comment (khancyr) [inline libraries/AP_MSP/AP_MSP.cpp:86]: if we don't use the protocol_instance variable, maybe using 

```suggestion
    while (_msp_status.backend_count < MSP_MAX_INSTANCES) {
```
would be better ?

File: libraries/AP_MSP/AP_MSP.cpp (hunk 1 above)
Comment (yaapu) [reply libraries/AP_MSP/AP_MSP.cpp:102]: mhmm I don't think so, shouldn't we check each serial protocol for all instances starting at 0 up to the MAX allowed?

File: libraries/AP_MSP/AP_MSP.cpp (hunk 2 above)
Comment (yaapu) [reply libraries/AP_MSP/AP_MSP.cpp:85]: @khancyr I refactored and removed backend_instance, now should be ok. It loops thru all instances per each serial protocol starting from instance 0 up to MAX  instances accounting for already found backends

**Review by tridge — APPROVED** [review]

**Pull request merged**